src/databricks/labs/ucx/workspace_access/scim.py

```diff
@@ -108,3 +108,24 @@ def _safe_get_group(self, group_id: str) -> Group | None:
         except NotFound:
             logger.warning(f"removed on backend: {group_id}")
             return None
+
+    @retried(on=[InternalError])
+    @rate_limited(max_requests=255, burst_period_seconds=60)
+    def _get_group_with_retries(self, group_id: str) -> iam.Group | None:
+        return self._ws.groups.get(group_id)
+
+    def _list_workspace_groups(self, scim_attributes: str) -> list[iam.Group]:
+        results = []
+        logger.info(f"Listing workspace groups with {scim_attributes}...")
+        # these attributes can get too large causing the api to timeout
+        # so we're fetching groups without these attributes first
+        # and then calling get on each of them to fetch all attributes
+        if "members" in scim_attributes or "roles" in scim_attributes or "entitlements" in scim_attributes:
+            for g in self._ws.groups.list(attributes="id,displayName,meta"):
+                group_with_all_attributes = self._get_group_with_retries(g.id)
+                results.append(group_with_all_attributes)
+        else:
+            for g in self._ws.groups.list(attributes=scim_attributes):
+                results.append(g)
```

Comment:
```suggestion
            results = self._ws.groups.list(attributes=scim_attributes)
```
for loop is redundant.

make sure to cover this branch with unit tests.